src/storage/leveldb/wallet_index.ts

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path from 'path';
+import { Level } from 'level';
+import { AbstractSublevel } from 'abstract-level';
+import { IKVWalletIndex, IWalletData, IWalletAccessData } from '../../types';
+import { GAP_LIMIT } from '../../constants';
+import { errorCodeOrNull, KEY_NOT_FOUND_CODE } from './errors';
+
+export const ACCESS_PREFIX = 'access';
+export const WALLET_PREFIX = 'wallet';
+export const GENERIC_PREFIX = 'generic';
+
+export default class LevelWalletIndex implements IKVWalletIndex {
+  dbpath: string;
+  accessDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, IWalletAccessData>;
+  walletDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, string>;
+  genericDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, any>;
+  indexVersion: string = '0.0.1';
+
+  constructor(dbpath: string) {
+    this.dbpath = path.join(dbpath, 'wallet');
+    const db = new Level(this.dbpath);
+    this.accessDB = db.sublevel<string, IWalletAccessData>(ACCESS_PREFIX, { valueEncoding: 'json' });
+    this.walletDB = db.sublevel(WALLET_PREFIX);
+    this.genericDB = db.sublevel<string, any>(GENERIC_PREFIX, { valueEncoding: 'json' });
+  }
+
+  async close(): Promise<void> {
+    await this.accessDB.db.close();
+  }
+
+  async _setNumber(dest: 'access'|'wallet'|'generic', key: string, value: number) {
+    const buf = Buffer.alloc(4);
+    buf.writeUInt32BE(value);
+    switch(dest) {
+      case 'access':
+        await this.accessDB.put<string, Buffer>(key, buf, { valueEncoding: 'buffer'});
+        break;
+      case 'wallet':
+        await this.walletDB.put<string, Buffer>(key, buf, { valueEncoding: 'buffer'});
+        break;
+      case 'generic':
+        await this.walletDB.put<string, Buffer>(key, buf, { valueEncoding: 'buffer'});
```

Comment:
```suggestion
        await this.genericDB.put<string, Buffer>(key, buf, { valueEncoding: 'buffer'});
```

Is this a bug? We should add a test for this.

Reply:
It is not a bug because this is not used.
The api defined to access the generic data only uses `getItem` and `setItem` with json encoding.

Removed `generic` from possibilities.